File: client/app/src/main/java/com/qubacy/geoqq/domain/mate/chat/model/MateMessage.kt
package com.qubacy.geoqq.domain.mate.chat.model

import com.qubacy.geoqq.data._common.model.message.DataMessage
import com.qubacy.geoqq.domain._common.model.user.User
import com.qubacy.geoqq.domain._common.model.user.toUser

data class MateMessage(
    val id: Long,
    val user: User,
    val text: String,
    val time: Long
) {

}

fun DataMessage.toMateMessage(): MateMessage {
    return MateMessage(id, user.toUser(), text, time)
}

File: server/own_services/common/pkg/messaging/geoqq/dto/payload/mateMessage.go
package payload

import (
	domain "common/pkg/domain/geoqq"
	"time"
)

type MateMessage struct {
	TargetUserId float64 `json:"target-user-id"`

	Id     float64 `json:"id"`
	ChatId float64 `json:"chat-id"`

	Text   string  `json:"text"`
	Time   float64 `json:"time"`
	UserId float64 `json:"user-id"` // from!
	Read   bool    `json:"read"`    // not used?
}

func (mm *MateMessage) ToDomain() *domain.MateMessageWithChat {
	return &domain.MateMessageWithChat{
		Id:     uint64(mm.Id),
		ChatId: uint64(mm.ChatId),
		Text:   mm.Text,
		Time:   time.Unix(int64(mm.Time), 0), // !
		UserId: uint64(mm.UserId),
		Read:   mm.Read,
	}
}
